recipes-core/data-partition/files/data.mount

```diff
@@ -0,0 +1,15 @@
+[Unit]
+Description=Mount /data partition
+Before=local-fs.target
+After=expand-on-first-boot.service
+Requires=systemd-fsck@dev-disk-by\x2dpartlabel-data.service
+After=expand-on-first-boot.service systemd-fsck@dev-disk-by\x2dpartlabel-data.service
+
+[Mount]
+What=/dev/disk/by-partlabel/data
+Where=/data
+Type=ext4
+Options=data=journal,commit=20,noatime,nodiratime,nodelalloc
```

Comment:
This would better be a separate Pull Request, this one here is as the title suggests about adding SWUpdate.

Aside, the mount options should be made configurable via a BitBake variable IMO, i.e., make this a `.tmpl`.